### ISSUE TITLE
Support manually providing a private IPv4 address for an internal network load balancer

### ISSUE THREAD

**Issue by arnogeurts** [terraform-providers/terraform-provider-aws#12329]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS supports manually providing a private IPv4 address to the NLB[1], however, this is not yet supported in this Terraform provider. It would be nice if the field "private_ipv4_address" would be added to the "subnet_mapping" block in the "aws_lb" resource.

### New or Affected Resource(s)

* aws_lb

### Potential Terraform Configuration

```hcl
resource "aws_lb" "lb_test" {
  name               = "my-network-load-balancer"
  load_balancer_type = "network"
  internal           = true

  subnet_mapping {
    subnet_id            = aws_subnet.private.id
    private_ipv4_address = "10.0.0.11"
  }
}
```

### References

[1] https://docs.aws.amazon.com/elasticloadbalancing/latest/network/network-load-balancers.html

* #0000

**Comment by ewbankkit** [comment]
Similar:
* #11044
* #11887
* #11403
* #11674

**Comment by bflad** [comment]
Hi folks 👋  This functionality has been merged and will release with version 3.1.0 of the Terraform AWS Provider, next Thursday. Thanks to @KadenLNelson for the implementation. 👍
